oidc-api/src/test/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandlerTest.java

```diff
@@ -216,12 +223,24 @@ class AuthorisationHandlerTest {
     public void setUp() {
         when(configService.getEnvironment()).thenReturn("test-env");
         when(configService.getDomainName()).thenReturn("auth.ida.digital.cabinet-office.gov.uk");
-        when(configService.getLoginURI()).thenReturn(LOGIN_URL);
         when(configService.getOrchestrationClientId()).thenReturn(TEST_ORCHESTRATOR_CLIENT_ID);
         when(configService.getSessionCookieAttributes()).thenReturn("Secure; HttpOnly;");
         when(configService.getSessionCookieMaxAge()).thenReturn(3600);
         when(configService.getPersistentCookieMaxAge()).thenReturn(34190000);
         when(configService.isIdentityEnabled()).thenReturn(true);
+        when(frontEndPages.baseURI()).thenReturn(FRONT_END_BASE_URI);
+        when(frontEndPages.errorURI()).thenReturn(FRONT_END_ERROR_URI);
+        when(frontEndPages.authorizeURI(any(), any()))
+                .thenAnswer(
+                        i -> {
+                            var queryParameters = new HashMap<String, String>();
+                            var prompt = (Optional<Prompt.Type>) i.getArgument(0);
+                            var googleAnalytics = (Optional<String>) i.getArgument(1);
+                            prompt.ifPresent(p -> queryParameters.put("prompt", p.toString()));
+                            googleAnalytics.ifPresent(
+                                    ga -> queryParameters.put("result", ga.toString()));
+                            return buildURI(FRONT_END_AUTHORIZE_URI, queryParameters);
+                        });
```

Comment:
Do we need this level of detail? If we really need it we could probably use a spy instead of a mock and delegate this call to the real class, but I suspect we could get away with a static return?

Reply:
Yeah, I agree, I remember thinking this wasn't great when I was writing it.